### ISSUE TITLE
Helping downstream projects to keep up with the change log

### ISSUE THREAD

**Issue by blaggacao** [goautodial/v4.0#53]
We have a downstream project which regularly fetches from upstream.

I was looking at the git log history and see things like:
![image](https://user-images.githubusercontent.com/7548295/97118950-1459a400-16db-11eb-8345-658d56a19a2a.png)

or things like 
```
commit fee58d07abb07a47f1802ecfb8bc454294c5f1de
Merge: c859a584 c2754363
Author: Chris <chris@iknowitworks.org>
Date:   Fri Oct 9 01:29:50 2020 +0800

    Merge branch 'master' of https://github.com/goautodial/v4.0
```

---

It would help if a series of commit messages could be squashed so that it becomes intelligible as a single atomic (working) change.

It is hard to follow how `Merge branch 'master' of https://github.com/goautodial/v4.0` does bring so may changes to `branch 'master' of https://github.com/goautodial/v4.0` (presumably the same). The only thing I could imagine is that somebody might have forced-pushed to master? &mdash; that would be unfortunate since it removes the very minimal git guaranties I would expect as a downstream user from a master branch.

I hope this report might prompt you guys to reconsider your branching/merging policies.

Thanks for this great project! I'll expect to be able to contribute some stuff in the near future.

**Comment by demianb** [comment]
Thank you for the feedback. We'll see what we can do to make it better for everyone. 

@mmixx please take note. Thanks!
